src/sentry/integrations/utils/code_mapping.py

```diff
@@ -299,36 +300,18 @@ def list_file_matches(self, frame_filename: FrameFilename) -> list[dict[str, str
             ]
 
             for file in matches:
+                stacktrace_root, source_path = find_roots(frame_filename.raw_path, file)
                 file_matches.append(
                     {
                         "filename": file,
                         "repo_name": repo_tree.repo.name,
                         "repo_branch": repo_tree.repo.branch,
-                        "stacktrace_root": f"{frame_filename.root}/",
-                        "source_path": _get_code_mapping_source_path(file, frame_filename),
+                        "stacktrace_root": stacktrace_root,
+                        "source_path": source_path,
```

Comment:
This is one of the places where we are deriving roots, so we once again replace the previous implementation with the `find_roots` function